presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushDownTableConstraints.java

```diff
@@ -143,7 +145,7 @@ private boolean planChanged(PlanNode rewrittenPlan, FilterNode oldPlan, Lookup l
         }
 
         FilterNode rewrittenFilter = (FilterNode) rewrittenPlan;
-        if (!rewrittenFilter.getPredicate().equals(oldPlan.getPredicate())) {
+        if (!ImmutableSet.copyOf(extractConjuncts(rewrittenFilter.getPredicate())).equals(ImmutableSet.copyOf(extractConjuncts(oldPlan.getPredicate())))) {
```

Comment:
Aren't there other cases. E.g. a condition rewritten to different shape (with different set of conjuncts) but same meaning.

E.g `(a<5) AND ((b<3) OR (c<3))` vs `(a<5) AND ((c<3) OR (b<3))`